src/main/java/org/example/PerformWordSearch.java

```diff
@@ -32,13 +31,16 @@ public void run() {
                     matchingResultsCount++;
                 }
             }
+            DatabaseHelper dbHelper = new DatabaseHelper();
             if (matchingResultsCount > 0) {
+                dbHelper.storingDataToDataBase(inputFilePath, searchWord, "Success", matchingResultsCount, "");
```

Comment:
This method storingDataToDataBase is called in both if and else with difference in 3rd and 5th parameters in respective places.  I suggest call this method only once before line number 42  . The 3rd and 5th parameters alone will be assigned within if and else conditions as per the situation and those parameters will be use in that method

Reply:
Made changes.